test/hotspot/jtreg/runtime/os/TestExplicitPageAllocation.java

```diff
@@ -0,0 +1,207 @@
+        dis = new DataInputStream(fis);
+        int pagecount = Integer.parseInt(dis.readLine());
+        dis.close();
+        fis.close();
+        return pagecount;
+    }
+
+    public static void doSetup() throws Exception {
+        // Large page sizes
+        File[] directories = new File(DIR_HUGE_PAGES).listFiles(File::isDirectory);
+        for (File dir : directories) {
+            String pageSizeFileName = dir.getName();
+            Matcher matcher = HUGEPAGE_PATTERN.matcher(pageSizeFileName);
+            matcher.find();
+            String pageSize = matcher.group(1);
+
+            if (pageSize != null) {
+                int freePageCount = checkAndReadFile(DIR_HUGE_PAGES+pageSizeFileName+"/free_hugepages", pageSize);
+                int resvPageCount = checkAndReadFile(DIR_HUGE_PAGES+pageSizeFileName+"/resv_hugepages", pageSize);
+
+                int availablePages = freePageCount - resvPageCount;
+                if (availablePages >= 0) {
+                    System.out.println("Number of available "+pageSize+"kB pages = "+availablePages);
+                } else {
+                    System.out.println("System does not support"+pageSize+"kB pages");
+                    continue;
+                }
+
+                int index = Integer.numberOfTrailingZeros(Integer.parseInt(pageSize)*1024);
+                pageSizes[index] = true;
+                pageCount[index] = availablePages;
+            }
+        }
+        // OS vm page size
+        ProcessBuilder processBuilder = new ProcessBuilder();
+        processBuilder.command("getconf", "PAGE_SIZE");
+        Process process = processBuilder.start();
+        StringBuilder output = new StringBuilder();
+        BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
+        String vmPageSize = reader.readLine();
+
+        vmPageSizeIndex = Integer.numberOfTrailingZeros(Integer.parseInt(vmPageSize));
+        pageSizes[vmPageSizeIndex] = true;
+        pageCount[vmPageSizeIndex] = Integer.MAX_VALUE;
+    }
+
+    public static void testCase(int index) throws Exception {
+        String pageSize = sizeFromIndex(index);
+        ProcessBuilder pb = ProcessTools.createJavaProcessBuilder("-Xlog:pagesize",
+                                                                  "-XX:LargePageSizeInBytes="+pageSize,
+                                                                  "-XX:+UseParallelGC",
+                                                                  "-XX:+UseLargePages",
+                                                                  "-Xmx2g",
+                                                                  "-Xms1g",
+                                                                  MyClass.class.getName());
+        OutputAnalyzer output = new OutputAnalyzer(pb.start());
+        output.shouldHaveExitValue(0);
+
+        for (int i = index;i >= vmPageSizeIndex;i--) {
+            if(pageSizes[i]) {
```

Comment:
An alternative to this would be to have something like `ArrayList<PageSizeConfig>` to which we only add the configured page sizes. And `PageSizeConfig` would hold both the page size and the number of pages configured for this size. I think that would be slightly easier to follow.

Reply:
Page Count array size increased to 64 this will fix the reported issue and should be sufficient to handle all the page sizes for 64 bit platforms.